atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -1200,6 +1223,24 @@ private static LockAndTimestampServices getLockAndTimestampServices(
                 .build();
     }
 
+    private static LeaderTimeGetter getLeaderTimeGetter(
+            String timelockNamespace,
+            Optional<TimeLockRequestBatcherProviders> timelockRequestBatcherProviders,
+            AtlasDbDialogueServiceProvider serviceProvider,
+            LeaderElectionReportingTimelockService namespacedConjureTimelockService) {
+        return timelockRequestBatcherProviders
+                .map(TimeLockRequestBatcherProviders::leaderTimeBatcherProvider)
+                .map(provider -> provider.getBatcher(getMultiClientTimelockServiceSupplier(serviceProvider)))
+                .<LeaderTimeGetter>map(batcher -> new NamespacedCoalescingLeaderTimeGetter(timelockNamespace, batcher))
```

Comment:
This broadly makes sense. As a matter of style I'd probably write this imperatively as there is a fair bit to take in, but this is fine if you prefer.